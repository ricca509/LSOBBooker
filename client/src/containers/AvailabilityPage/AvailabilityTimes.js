import React from 'react';
import moment from 'moment';

const AvailabilityTimes = ({ times }) => (
  <span className="row">
    { times.map(time => (
      <a key={time}
        target="blank"
        className="column"
        href="http://londonschoolofbarbering.simplybook.me/sheduler/manage/category/4/event/68">
        {moment(time, 'HH:mm:ss').format('HH:mm')}
      </a>
    )) }
  </span>
);

AvailabilityTimes.displayName = 'AvailabilityTimes';

export default AvailabilityTimes;
