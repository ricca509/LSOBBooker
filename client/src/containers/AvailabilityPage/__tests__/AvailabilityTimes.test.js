import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import AvailabilityTimes from '../AvailabilityTimes';

describe('AvailabilityTimes', () => {
  const times = [
    '12:30:00',
    '15:00:00',
    '20:00:00'
  ];

  it('renders a list of times with links', () => {
    const wrapper = shallow(<AvailabilityTimes times={times} />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
