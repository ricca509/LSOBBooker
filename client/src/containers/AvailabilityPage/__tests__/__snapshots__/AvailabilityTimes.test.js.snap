exports[`AvailabilityTimes renders a list of times with links 1`] = `
<ul>
  <li>
    <a
      href="http://londonschoolofbarbering.simplybook.me/sheduler/manage/category/4/event/68"
      target="blank">
      12:30:00
    </a>
  </li>
  <li>
    <a
      href="http://londonschoolofbarbering.simplybook.me/sheduler/manage/category/4/event/68"
      target="blank">
      15:00:00
    </a>
  </li>
  <li>
    <a
      href="http://londonschoolofbarbering.simplybook.me/sheduler/manage/category/4/event/68"
      target="blank">
      20:00:00
    </a>
  </li>
</ul>
`;
